src/detect-dce-iface.c

```diff
@@ -180,8 +180,12 @@ static DetectDceIfaceData *DetectDceIfaceArgParse(const char *arg)
             goto error;
         }
 
-        version = atoi(copy_str);
-        if (version > UINT16_MAX) {
+        if (ByteExtractStringInt32(&version, 10, 0, (const char *)copy_str) < 0) {
+            SCLogError(SC_ERR_INVALID_SIGNATURE, "DCE_IFACE interface version "
+                       "invalid: \"%s\n\"", copy_str);
+            goto error;
```

Comment:
IMO, the code is now stricter so the test should be revisted. If the purpose of the test is to `assert version == 0` (which would've happened previously with atoi, then the test can be reworked.

Reply:
Turns out the test does a lot of things before checking if the version is 0.
I have understood the following.

**Aim of the test**
To check that uuid, version and options are parsed correctly from a given argument string which does not have a version specified.

**Procedure**
The function calls are as follows.
```
DetectDceIfaceTestParse06    ---->     DetectDceIfaceSetup   ---->    DetectDceIfaceArgParse
```
1. Since we `goto error` from the middle of the function `DetectDceIfaceArgParse`, the return value is `NULL` compared to a `DetectDceIfaceData *` under normal circumstances.
2. Since `DetectDceIfaceSetup` receives `NULL` from `DetectDceIfaceArgParse`, it returns `-1` (which is an indicator of failure) right away without doing a `SigMatch` allocation. `SigMatch` allocation and list entry happens if there was no error constructing `DetectDceIfaceData` struct.
3. The test as of now tends to fail because of (2) but it should ideally fail since we set `did` to `NULL` and there is nothing to check the `test_uuid` against. So, the entire point of test and keyword kinda fails.

**Solution (My proposal)**
Change the test argument string to something with version as that is not the sole purpose of this test.

**Caveat**
By putting this error condition we are enforcing the presence of version with dce_iface string which does not go as per docs or `PARSE_REGEX`. (As per docs and regex only `uuid` is required, `version` and `any_frag` option can be skipped)

**Question**
Do we want to enforce `version` in the `dce_iface` argument string? If yes, we should update the test, docs and regex. If no, version should be set to `0` in case of an invalid/ version string.

Reply:
Yes. Just to double confirm, with the patch above, we are enforcing that we are not accepting argument string to `dce-iface` of the type `"12345678-1234-1234-1234-123456789ABC,any_frag"`, I hope its OK?
Acceptable strings can be:
`"12345678-1234-1234-1234-123456789ABC"`
`"12345678-1234-1234-1234-123456789ABC,<1,any_frag"`
`"12345678-1234-1234-1234-123456789ABC,<1"`
only.

Reply:
I think I found a way to fix this. Thanks for all the assistance with this PR.

Reply:
```12345678-1234-1234-1234-123456789ABC,any_frag``` notation is used in 2 ETPRO rules. I can share them privately.